src/lib/core/common-behaviors/color.ts

```diff
@@ -0,0 +1,47 @@
+import {Constructor} from './constructor';
+import {ElementRef, Renderer2} from '@angular/core';
+
+/** @docs-private */
+export interface CanColor {
+  color: string;
+}
+
+/** @docs-private */
+export interface HasRenderer {
+  _renderer: Renderer2;
+  _elementRef: ElementRef;
+}
+
+/** Possible color palette values.  */
+export type ThemePalette = 'primary' | 'accent' | 'warn' | null;
+
+/** Mixin to augment a directive with a `color` property. */
+export function mixinColor<T extends Constructor<HasRenderer>>(base: T, allowNoColor = false)
+    : Constructor<CanColor> & T {
+  return class extends base {
+    private _color: ThemePalette = undefined;
+
+    constructor(...args: any[]) { super(...args); }
+
+    get color(): ThemePalette { return this._color; }
+    set color(value: ThemePalette) {
+      if (value || allowNoColor && !value) {
```

Comment:
You mean that we can pass a value that is the default value to the mixin? (e.g `null`)

I'd rather do it the other way around because the default value "should" be basically everything that is falsy.